### ISSUE TITLE
API docs missing DefaultContextInteractions; an encouraging word

### ISSUE THREAD

**Issue by djradon** [sarl/sarl#403]
I assume that DefaultContextInteractions has been removed from 0.4 since they're no longer present in the API docs, and when I try to follow the tutorials, Eclipse complains about it. Also can't get the pre-configured eclipse to work, but I'm not really looking for specific advice. This "new issue" is really more "encouragement". 

Also curious about time management in the new platform. The old Janus had ConstantKernelTimeManager, but I can't find a new model of time management in 0.3 or 0.4.

I can't find a developer mailing list for Sarl or Janus. If there were one, I'd like to join. Maybe you attract more contributors by opening up your process more?

I've been reviewing open-source MAS platforms, and I think this new holonic direction for sarl/janus is very promising. I also love the use of modern programming techniques and libraries. But I have had a lot of trouble with maven build of sarl-demos, eclipse integration, matching SRE to proper release and getting tutorials to work.

My guess is y'all are juggling tight resources to keep this project moving forward. I wish you all the best in 2016.

**Comment by djradon** [comment]
BTW, from the README.md: "it can be linked with other existing agent platforms and frameworks"... mark me +1 for Jade integration.

**Comment by gallandarakhneorg** [comment]
Thank you for your message. It is nice to see your interest for our language and our platform.

`DefaultContextInteractions` should be in the API documentation. So, I push up the issue #404 for solving this problem.

You're right, `ConstantKernelTimeManager` is not in the SARL language API nor in the new version of Janus. Because this new version of Janus is running all the elements in parallel, it becomes difficult to have a synchronized execution of the agents. Consequently, it becomes also difficult to determine when it is required to make the time evolving. For these reasons, the `*TimeManager` classes from the version 1 of Janus do not exist in the version 2. Nevertheless, I think synchronized execution process may be helpful for simulation purposes. That's why I'm exploring some ways for having this execution mode, and a time manager, with SARL and Janus v2. The related projects are: i) [a simulator of agents moving on a plane](https://github.com/gallandarakhneorg/vi51-student-sarl-simulation) ii) [the pacman game](https://github.com/gallandarakhneorg/ia54-student-sarl-simulation), and iii) a [complete 2D grid environment for SARL](https://github.com/gallandarakhneorg/jaak). These projects have each advantages and drawbacks. When I think something stable is ready, I will propose it as a plugin for Janus/SARL. You're welcome to give a feedback on them :smile: 

Regarding the mailing list, we have not set up yet. I pushed the issue #405 for creating one.

Most of the troubles with our environment are due to different versions of the SARL SDK in the Eclipse IDE and in the project dependencies (including the Maven dependencies).
They must be the same. Otherwise, it's becomes crazy.
If SARL has the version `x.y.z`, then Janus must have the version `2.x.y.w`.
We plan to include Janus and demos in the Eclipse IDE for solving this project (issues #303 and #314).
Additionally, I discovered a major bug in the Maven plugin for SARL. It should be fixed in version 0.3.1 (issue #402). If you still have problems, please open an issue with the error messages. We will manage it asap.

Regarding your comment about Jade and our tools, are you linking Janus and Jade, or generating Jade code from SARL code? We have considered mostly this second point in the README. But the first is very interesting too. Have you some repository that includes your proposal? It may be interesting to put its link in [awesome-sarl](https://github.com/sarl/awesome-sarl).

Our resources are limited indeed. Contributors are welcome :smile:
